### ISSUE TITLE
Accessibility Review

### ISSUE THREAD

**Issue by samikeijonen** [tomusborne/generatepress#35]
Here is the first list of [accessibility review](https://make.wordpress.org/themes/handbook/review/accessibility/required/).

## Keyboard Navigation

Test using Tab and Shift + Tab and check can you see where you're navigating.

- In general all the links are missing good focus styles. There is color change from blue to black but even that doesn't have enough [color contrast](http://leaverou.github.io/contrast-ratio/#%23000-on-%231e73be) (1: 4.5).
- In post author name, category/tags/next/previous post links there is no color change on focus.
- I suggest that there would be some kind of outline when links are focused. 
- Usually color alone is not good way to style focus. It can be hard for color blinds for example.
- In primary menu there is also same focus style issue but it's probably not that critical. But I suggest to have outline focus styles in there also. 
- In Firefox form fields focus styles could be better. I can see only subtle background-color change.
- Form submit button (or other buttons) also have only color change on focus and doesn't meet color contrast criteria.
- On mobile menu-toggle button doesn't have focus styles at all.
- On mobile you can't access to dropdown menus using keyboard. `dropdown-menu-toggle` should be a `button`, not a `span`. Button would have native keyboard support.

## Controls

- From previous `dropdown-menu-toggle` should be a `button`, not a `span`.

## Skip Links

- Pass
- Note that in IE11 skip link is broken, that's why we usually use [skip-link-focus.js](https://github.com/Automattic/_s/blob/master/js/skip-link-focus-fix.js) fix.

## Forms

- There are no labels in the comment form. Labels can be visually hidden, [see examples](https://make.wordpress.org/themes/handbook/review/accessibility/required/#forms).

## Headings

- No big issues but widget titles should probably be `h2`, not `h4`. Now heading hierarchy skips heading levels from `h1` or `h2` to `h4`.

## ARIA Landmark Roles

- All content should be inside main elements. For example footer widgets are not inside `footer`.
- Even if we removed ARIA landmark roles we might add them back in due to recent changes. See [ARIA landmark roles section](https://make.wordpress.org/themes/handbook/review/accessibility/required/#aria-landmark-roles).

## Link Text

- Read more -link is called repetitive link text and should be avoided. Short [example in the guidelines](https://make.wordpress.org/themes/handbook/review/accessibility/required/#link-text).

## Contrasts

- Pass. At least I couldn't find any big issues:)

## Images

- Feature image on blog archives should have `alt` attribute, not `title` attribute.

## Screen Reader Text

- Consider updating `screen-reader-text` class: https://github.com/Automattic/_s/blob/master/style.css#L656

**Comment by tomusborne** [comment]
Thank you for this!

I've made some changes to the keyboard navigation and focus styles over in the vanilla JS branch. 

The Firefox form issue is strange - do they not have a standard outline on forms?

Turning `dropdown-menu-toggle` may result in some styling/back-compat issues, but I did add `tabindex` to the `span` elements for now, so they're at least accessible using the tab key. I'll look into turning them into button elements in the near future.

Moving the footer widgets into the footer element might cause some back compat/existing child theme issues as well - I'll play with that.

Will continue on with your other suggestions.

**Comment by tomusborne** [comment]
Images and Link Text have also been dealt with now.

**Comment by samikeijonen** [comment]
Here is second round for `a11y` review.

## Keyboard navigation and forms
- `menu-toggle` button doesn't toggle `aria-expanded` state to `true` and `false` when opened and closed.
- Same issue with `dropdown-menu-toggle` button. Also it should be a `<button>`, not `<span>`.  There is also label missing for that button. It can inside `screen-reader-text` class. Something like `<button class="dropdown-menu-toggle"><span class="screen-reader-text">Child menu</span></button>`. I think you can keep the styling similar even when changing markup.
- There are still no labels in the comment form.

## Headings
- Widget titles should be `<h2>`.
- There is no `<h1>` in "Blog" page (when front page is regular page, and post page something else). In [`_s` we add it it like this](https://github.com/Automattic/_s/blob/master/index.php#L23). 

## Aria landmarks
- Aria landmarks are still a little bit of and all content should be inside a landmark. Understand if it's going to break child themes and is a no go.
- New issue about sidebar markup. I noticed that Voiceover repeats `Complementary` landmark in every widget, it should be announced only once. Markup should probably be like this:

```html+php
<aside role="complementary">
	<section class="widget">
	<section class="widget">
	<section class="widget">
</aside>
```

## Screen Reader Text
- Update the `screen-reader-text` class. 

## Contrasts

- `entry-meta` color is `#888` which is too low on white background. Same color might be somewhere else also.

## Images
- You can remove `<title>` in featured images now that there is `alt`.

**Comment by tomusborne** [comment]
Just an update to this issue. The only outstanding issue is the dropdown menu selector, which we're tackling in 2.2.
